### PR TITLE
Extend DynamicConsoleLoggerProvider to provide dynamic levels for Serilog 

### LINKED ISSUE #685: Dynamic Serilog: Unable to increase log level beyond Default
## Describe the bug
When using Serilog dynamic logging integration, unable to increase logging verbosity beyond `Default` logger level in app configuration.

## Steps to reproduce 
1. Configure Dynamic Serilog
2. Configure Loggers actuator
3. Set `Default` logger at "Information" level
4. Invoke logger for each level 
  a. should log at Info/Warn/Error/Critical
5. Invoke `POST` loggers actuator to change `Default` logger to "Trace" level. 
  a. Confirm `effectiveLevel` via `GET` loggers actuator.
6. Invoke logger for each level
  a. Expected: should log at all levels
  b. Actual: still only logs at Info/Warn/Error/Critical

## Expected behavior
Should be able to run application with `Default` logger level at "Information", then dynamically increase logging verbosity at runtime for issue investigation.

## Environment (please complete the following information):
 - Steeltoe Version: 2.4.5 / 3.0.2 / 3.1.0-rc1
 - Platform: local development environment
 - OS: Windows
 - .NET Version: net5.0
 - Any other library versions to note
   - Serilog Version: transitive (no direct dependencies)

## Additional context or links

Here is a minimal example. Invoking the scaffolded `GET /weatherforecast` endpoint will call the logger for each available level. The loggers actuator is at `/actuator/loggers`. There are a few branches testing various workarounds.
* https://github.com/jeremyhayes/steeltoe-dynamic-serilog-example

As a workaround, you can set the `Default` logger to `"Trace"` level. It is important to note that this must be done via the _Microsoft_ logger config, regardless of _Serilog's_ configuration. The tradeoff here is that you must now provide overrides for any/all categories which shouldn't use Verbose logging, which can be unwieldy depending on included packages.

This does not affect the non-Serilog `DynamicConsoleLogger`.

I suspect this is an issue with Serilog's `LoggingLevelSwitch`, but I'm not sure if it's due to Steeltoe's usage, or in Serilog itself. Figured I'd start here. :)

### PR DESCRIPTION
**Pull request by hananiel**
Replaces IloggerFactory with SerilogLogger factory. Resolves #685 , #662, #478. 
Completely reworked serilog to use DynamicConsoleLogger and proxying to Serilog Logger via delegate. This change also uses the `
`DynamicConsoleLoggerProvider` implementation for filtering logs instead of using Serilog's level switches to provide dynamic changes to log levels by namespace.

### CODE REVIEW

File: src/Logging/src/DynamicSerilogCore/SerilogWebHostBuilderExtensions.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ public static class SerilogWebHostBuilderExtensions
         /// </summary>
         /// <param name="hostBuilder">The <see cref="IWebHostBuilder"/> to configure</param>
         /// <param name="configureLogger">The delegate for configuring the <see cref="DynamicLoggerConfiguration" /> that will be used to construct a <see cref="Serilog.Core.Logger" /></param>
-        /// <param name="preserveStaticLogger">Indicates whether to preserve the value of <see cref="Log.Logger"/>.</param>
+        /// <param name="preserveStaticLogger">Not Supported!</param>
```
Comment (hananiel) [inline src/Logging/src/DynamicSerilogCore/SerilogWebHostBuilderExtensions.cs:18]: Although we gave this option, it never worked - cannot use static logger and dyanmically allow setting logs.

Comment (TimHess) [reply src/Logging/src/DynamicSerilogCore/SerilogWebHostBuilderExtensions.cs:18]: If that's the case we should probably mark this signature obsolete and provide a more accurate option

File: src/Logging/src/DynamicSerilogBase/SerilogLoggingBuilderExtensions.cs [hunk 2]
```diff
@@ -6,69 +6,81 @@
...
+                builder.AddFilter<SerilogDynamicProvider>(null, LogLevel.Trace); // TODO : What does this do?
+                builder.Services.AddOptions();
+
+                if (serilogConfiguration != null)
+                {
+                    builder.Services.Configure<SerilogOptions>(options => options.SetSerilogOptions(serilogConfiguration));
+                }
+                else
+                {
+                    builder.Services.Configure<SerilogOptions>(options => options.SetSerilogOptions(configuration));
```
Comment (TimHess) [inline src/Logging/src/DynamicSerilogBase/SerilogLoggingBuilderExtensions.cs:47]:
```suggestion
                    builder.Services
                        .AddOptions<SerilogOptions>()
                        .Configure<IConfiguration>((options, configuration) => options.SetSerilogOptions(configuration));
```
With this code we can avoid the usage of BuildServiceProvider to get the IConfiguration

File: src/Logging/src/DynamicSerilogCore/SerilogWebHostBuilderExtensions.cs (hunk 1 above)
Comment (hananiel) [reply src/Logging/src/DynamicSerilogCore/SerilogWebHostBuilderExtensions.cs]: Fixed.

File: src/Logging/src/DynamicSerilogBase/SerilogLoggingBuilderExtensions.cs (hunk 2 above)
Comment (hananiel) [reply src/Logging/src/DynamicSerilogBase/SerilogLoggingBuilderExtensions.cs]: Thanks for suggestion, it is working well.

**Comment by TimHess** [comment]
Consider renaming DynamicConsoleLogger

**Comment by TimHess** [comment]
Update description on DynamicSerilogBase to remove `Includes Console sink.`

File: src/Logging/src/DynamicSerilogBase/ISerilogOptions.cs [hunk 3]
```diff
@@ -12,8 +13,10 @@ public interface ISerilogOptions
 
         MinimumLevel MinimumLevel { get; set; }
 
+        [Obsolete("No longer needed with current implementation. Will be removed in next major release")]
```
[the remark this answers is not in the document]
Comment (hananiel) [reply src/Logging/src/DynamicSerilogBase/ISerilogOptions.cs:16]: No they do not. Before it was an opt in to exclude configuration you want our implementation to ignore, now it is all ignored for the options we use (override levels) and all applied for all serilog components.

File: src/Logging/src/DynamicSerilogBase/SerilogHostBuilderExtensions.cs [hunk 4]
```diff
@@ -27,13 +27,15 @@ public static class SerilogHostBuilderExtensions
             return hostBuilder
                 .ConfigureLogging((hostContext, logBuilder) =>
                 {
-                    var loggerConfiguration = SerilogConfigurationExtensions.GetDefaultSerilogConfiguration(hostContext.Configuration);
+                    LoggerConfiguration loggerConfiguration = null;
+
                     if (configureLogger is object)
                     {
+                        loggerConfiguration = new LoggerConfiguration().ReadFrom.Configuration(hostContext.Configuration);
```
[the remark this answers is not in the document]
Comment (hananiel) [reply src/Logging/src/DynamicSerilogBase/SerilogHostBuilderExtensions.cs:34]: This extension was added to allow an end user to set somethings from HostContext in configuring serilog. In the LoggingBuilder we already have a parameter to pass serilog Configuration, so they can do whatever they want there before passing it in.

File: src/Logging/src/DynamicSerilogBase/SerilogConfigurationExtensions.cs [hunk 5]
```diff
@@ -14,13 +14,23 @@ internal static class SerilogConfigurationExtensions
         /// </summary>
         /// <param name="configuration"><see cref="IConfiguration"/></param>
         /// <returns>What Steeltoe considers to be a default <see cref="LoggerConfiguration" /></returns>
-        internal static LoggerConfiguration GetDefaultSerilogConfiguration(IConfiguration configuration) => new LoggerConfiguration().ReadFrom.Configuration(configuration).WriteToConsole();
+        internal static LoggerConfiguration GetDefaultSerilogConfiguration(IConfiguration configuration) => new LoggerConfiguration().ReadFrom.Configuration(configuration);
```
[the remark this answers is not in the document]
Comment (hananiel) [reply src/Logging/src/DynamicSerilogBase/SerilogConfigurationExtensions.cs:17]: agreed

File: src/Logging/src/DynamicSerilogBase/SerilogOptions.cs [hunk 6]
```diff
@@ -16,34 +17,70 @@ public class SerilogOptions : ISerilogOptions
...
             var section = configuration.GetSection(ConfigPath);
             section.Bind(this);
             if (MinimumLevel == null)
             {
+                var defaultLevel = LogEventLevel.Information;
+
+                var strMinLevel = section.GetValue<string>("MinimumLevel");
+                if (!string.IsNullOrEmpty(strMinLevel))
+                {
+                    Enum.TryParse(strMinLevel, out defaultLevel);
```
[the remark this answers is not in the document]
Comment (hananiel) [reply src/Logging/src/DynamicSerilogBase/SerilogOptions.cs:39]: added a test

**Review by TimHess — APPROVED** [review]

**Comment by hananiel** [comment]
Waiting for 3.1.1 to be released to merge this PR.

**Pull request merged**